proxy/grpcproxy/watch.go

```diff
@@ -192,6 +201,22 @@ func (wps *watchProxyStream) close() {
 	close(wps.watchCh)
 }
 
+func (wps *watchProxyStream) checkPermissionForWatch(key, rangeEnd []byte) error {
+	if len(key) == 0 {
+		// If the length of the key is 0, we need to obtain full range.
+		// look at clientv3.WithPrefix()
+		key = []byte{0}
+		rangeEnd = []byte{0}
+	}
+	req := &pb.RangeRequest{
+		Serializable: true,
+		Key:          key,
+		RangeEnd:     rangeEnd,
+	}
```

Comment:
`CountOnly: 1, Limit: 1` so this doesn't fetch a lot of data

Reply:
Thanks, I'll update for the comments soon.